presto-main/src/main/java/com/facebook/presto/execution/scheduler/group/DynamicLifespanScheduler.java

```diff
@@ -162,20 +187,23 @@ public SettableFuture schedule(SourceScheduler scheduler)
     {
         // Return a new future even if newDriverGroupReady has not finished.
         // Returning the same SettableFuture instance could lead to ListenableFuture retaining too many listener objects.
+        // todo resolve availableTasks being limited for unlimited execution
```

Comment:
Let's make it clearer, for example:

TODO: After initial scheduling, tasks would only be available after they finished at least one bucket. This is not necessarily the case if the initial scheduling covered all buckets and the available slots is not fully utilized (concurrentLifespansPerTask is large or infinite). In this case if a task failed, the recovered driver groups have to wait for tasks to be available again after finishing at least one bucket, even though by definition of concurrentLifespansPerTask they are already available.